src/ui/server/controllers/redirects/index.test.ts

```diff
@@ -12,12 +15,33 @@ describe('controllers/redirects', () => {
     res = mockRes();
   });
 
-  it('should redirect to a new URL via generateRedirectUrl helper function', () => {
-    get(req, res);
+  describe('when the req.original is valid', () => {
+    beforeEach(() => {
+      req = mockReq();
+    });
+
+    it('should redirect to a new URL via generateRedirectUrl helper function', () => {
+      get(req, res);
+
+      const expected = generateRedirectUrl(req.originalUrl);
+
+      expect(res.redirect).toHaveBeenCalledTimes(1);
+      expect(res.redirect).toHaveBeenCalledWith(expected);
+    });
+  });
+
+  describe('when the req.original is invalid', () => {
+    beforeEach(() => {
+      const invalid = { forgery: true };
+      // @ts-ignore
+      req.originalUrl = invalid;
```

Comment:
Why do we need to do this, with the URL being an object and `ts-ignore`? Express controls `originalUrl`